src/fontlibc/fontlibc.h

```diff
@@ -0,0 +1,437 @@
+ */
+void fontlib_SetNewlineCode(char code_point);
+
+/**
+ * Returns the code point that is currently recognized as being a newline.
+ * @return Current newline
+ */
+char fontlib_GetNewlineCode(void);
+
+/**
+ * Sets an alternate code point to recognize as a stop code.
+ * For example, you can set this to space to make DrawString and GetStringWidth
+ * stop processing when they reach a space.
+ * Setting this to zero will cause this to be ignored.
+ * NULL (0) will still be recognized as a stop code regardless of value.
+ * Defaults to 0.
+ * @param code_point Additional code point to recognize as a stop code.
+ */
+void fontlib_SetAlternateStopCode(char code_point);
+
+/**
+ * Returns the current alternate stop code.
+ * @return Current alternate stop code.
+ */
+char fontlib_GetAlternateStopCode(void);
+
+/**
+ * Sets the first code point considered printable.
+ * All code points before this will be considered control codes.
+ * Setting this to 0 (NULL) will NOT cause NULL to be ignored.
+ * This defaults 0x10.
+ * @param code_point First printable code point
+ */
+void fontlib_SetFirstPrintableCodePoint(char code_point);
+
+/**
+ * Returns the first code point considered printable.
+ * @return Current first printable code point
+ */
+char fontlib_GetFirstPrintableCodePoint(void);
+
+/**
+ * Returns the width of the given glyph
+ * @param codepoint Codepoint to test
+ * @return Width of glyph, 0 if invalid codepoint
+ */
+uint8_t fontlib_GetGlyphWidth(char codepoint);
+
+/**
+ * Returns the width of a string printed in the current font.
+ * Stops processing when it encounters ANY control code or a codepoint not in
+ * the current font.
+ * @param str Pointer to string
+ * @return Width of string
+ */
+uint24_t fontlib_GetStringWidth(char* str);
+
+/**
+ * Returns the width of a string printed in the current font.
+ * Stops processing when it encounters ANY control code or a codepoint not in
+ * the current font, or when max_characters have been processed.
+ * @param str Pointer to string
+ * @param max_characters Maximum number of characters to process
+ * (or 0 if not needed)
+ * @return Width of string
+ */
+uint24_t fontlib_GetStringWidthL(char* str, int24_t max_characters);
+
+/**
+ * Gets the location of the last character processed by GetStringWidth or 
+ * DrawString
+ * @return Pointer to character
+ */
+char* fontlib_GetLastCharacterRead(void);
+
+/**
+ * Returns 0 if DrawStringL or GetStringWidthL returned because max_characters
+ * were processed.  Returns a non-zero number otherwise.
+ * @return Either zero, or a non-zero number depending on the reason DrawStringL
+ * or GetStringWidthL returned.
+ */
+int24_t fontlib_GetCharactersRemaining(void);
+
+/**
+ * Draws a glyph
+ * @param glyph Codepoint
+ */
+void fontlib_DrawGlyph(uint8_t glyph);
+
+/**
+ * Draws a string
+ * THIS IS NOT REENTRANT (though if you need that, you're probably not using C)
+ * @param str Pointer to string
+ */
+void fontlib_DrawString(char* str);
```

Comment:
Shouldn't that be `const char* str`?

Reply:
const char *str. Anyone who ties the * to the type and not the variable should be stabbed.